source/adapters/level_zero/queue.cpp

```diff
@@ -599,7 +599,7 @@ ur_result_t ur_queue_handle_legacy_t_::queueRelease() {
       // internal reference count. When the External Reference count == 0, then
       // cleanup of the queue begins and the final decrement of the internal
       // reference count is completed.
-      Queue->RefCount.decrementAndTest();
+      static_cast<void>(Queue->RefCount.decrementAndTest());
```

Comment:
Perhaps we should assert that the return value is 0 here?

Reply:
No, because that would imply the refcnt "must" have been ==1 , which is not the case.

What occurs here is that the ExternalRef != 0 after being decremented. This means that the internal refcnt also needs to be decremented, but it may not be 0 so the refcnt being decremented, but not 0 is not a failure.

Reply:
In this case we don't care that the refcnt is not 0 at this time, we just need to decrement the atomic refcnt.